Easydict/NewApp/View/SettingView/Tabs/ServiceTab.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  ServiceTab.swift
+//  Easydict
+//
+//  Created by phlpsong on 2024/1/6.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import SwiftUI
+
+@available(macOS 13, *)
+struct ServiceTab: View {
+    @State private var windowTypeValue = EZWindowType.mini.rawValue
+    @State private var serviceTypes: [ServiceType] = []
+    @State private var services: [QueryService] = []
+    @State private var selectedIndex: Int?
+
+    var segmentCtrl: some View {
+        Picker("", selection: $windowTypeValue) {
+            Text("mini_window")
+                .tag(EZWindowType.mini.rawValue)
+
+            Text("fixed_window")
+                .tag(EZWindowType.fixed.rawValue)
+
+            Text("main_window")
+                .tag(EZWindowType.main.rawValue)
+        }
+        .padding()
+        .pickerStyle(.segmented)
+        .onChange(of: windowTypeValue) { type in
+            loadService(type: type)
+        }
+    }
+
+    var serviceList: some View {
+        List {
+            ForEach(Array(serviceTypes.enumerated()), id: \.offset) { index, _ in
```

Comment:
`ForEach(Array(serviceTypes.enumerated()), id: \.offset)`

This is not the proper usage of `ForEach` in SwiftUI.

Reply:
Yes, use the `zip` operator instead.

Reply:
No. You may have a misunderstanding of `ForEach` + `id`.

This is a common mistake people make in SwiftUI.

`offset` should not be your id unless you are taking some static data. You have take a look at `Identifiable` and `ForEach` doc.

Also, some great stackoverflow post also explains this well. I don't want to repeat the reason here.